### ISSUE TITLE
Oauth2 Authorization redirect_uri_mismatch

### ISSUE THREAD

**Issue by jonecx** [tumblr/docs#122]
In Android WebView, trying to implement a third party app authentication for users. My request url is

Note: the parameter values are replaced with fake values for the purpose of this question

https://www.tumblr.com/oauth2/authorize?client_id=T0G9la1U2QUYJnUM6wuYEWinQWx8gDRS04KB6YtX4r&response_type=code&scope=write&redirect_uri=hik://authorized.com&state=OjE3MDcyNDAxxMzA6akZTSXBmY2k4dXB6YTVHTWdmR1JaY2hVTWRZ

it has all the values client_id, response_type, redirect_uri, state and scope. Just like in the [API description](https://www.tumblr.com/docs/en/api/v2#oauth2authorize---authorization-request)

It opens the sign in page, it lets them sign in and shows them allow/deny page but when users hit the "Allow" button

it show show them "hiq://authorized.com?error=redirect_url_mismatch&error_description=The+redirect+URI+provided+is+missing+or+does+not+match&error_uri=http..."

I have the same redirect URI in my request and the Tumblr Apps Console which

Default callback URL:hik://authorized.com
OAuth2 redirect URLs (space separate):hik://authorized.com

I am doing exactly like in the documentation but no luck so far

**Comment by sanmai** [comment]
It hard to tell what is wrong without looking at the actual at the actual app ID and redirects.

Please consider using Tumblr Support form if you can't share them openly. 

https://tumblr.com/support

**Comment by jonecx** [comment]
@sanmai When you said the actual app ID, did you mean the actual client_id (OAuth Consumer Key) or the Application Name in the ```https://www.tumblr.com/oauth/apps``` forms?

**Comment by sanmai** [comment]
Having a client_id will certainly help.

**Comment by jonecx** [comment]
Thanks for the response, when I add a trailing slash to 
` url= "https://www.tumblr.com/oauth2/authorize?
        client_id=T0G9la1U2QUYXAbeDQu5JnUM6wuYEWinQWx8gDRS04KB6YtX4r&
        response_type=code&
        scope=write&
        redirect_uri=qio://authorized.com/`
I get an error like this 
qio://authorized.com/?error=invalid_request&error_description=The+state+parameter+is+required#_=_

I added a trailing / at the tumblr/auth/apps console and didn't make any difference.

**Comment by jonecx** [comment]
That is true :) I am sorry about that. Yeah I have got the code now. Thanks for the help

For documentation purposes this is how the url has to be in the tumblr/oauth/apps console
- Default Callback URl should have a trailing slash
- Oauth 2 redirect url should have a trailing slash
